File: app/src/main/java/mastermind/Peg.java
package mastermind;

import java.util.Objects;

public class Peg {

    private final int pColor;
    private boolean active;

    public Peg(int pegcolor){
        pColor = pegcolor;
        active = true;
    }

    public Peg(int pegcolor, boolean pickable){
        pColor = pegcolor;
        this.active = pickable;
    }

    public int getpColor() {
        return pColor;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Peg peg = (Peg) o;
        return pColor == peg.pColor;
    }

    @Override
    public int hashCode() {
        return Objects.hash(pColor);
    }
}

File: app/src/main/kotlin/mastermind/Peg.kt
package mastermind

import java.util.*

class Peg {
    private val pColor: Int
    var isActive: Boolean

    constructor(pegcolor: Int) {
        pColor = pegcolor
        isActive = true
    }

    constructor(pegcolor: Int, pickable: Boolean) {
        pColor = pegcolor
        isActive = pickable
    }

    fun getpColor(): Int {
        return pColor
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val peg = other as Peg
        return pColor == peg.pColor
    }

    override fun hashCode(): Int {
        return Objects.hash(pColor)
    }
}